### ISSUE TITLE
Explore hatch vs. nox for managing high level developer workflows

### ISSUE THREAD

**Issue by matthewfeickert** [scikit-hep/pyhf#2096]
Need to decide if we can use hatch envs with the current workflows that we have around nox (which I don't want to change in this PR) or if we'll need to remove hatch envs for the time being and make an Issue to decide how to move forward using `nox` and `hatch` in the future.

_Originally posted by @matthewfeickert in https://github.com/scikit-hep/pyhf/pull/2095#discussion_r1080896032_

**Comment by ofek** [comment]
Let me know if you have any questions! Environment management and configuration is probably the feature I put the most thought and design work into so I'm interested to see how this goes 🙂

**Comment by matthewfeickert** [comment]
Based on https://github.com/scikit-hep/pyhf/pull/2118#issuecomment-1432584212 @kratsg and @agoose77 do you have any clever thoughts on how to determine if `tensorflow` or `tensorflow-macos` should be installed at runtime depending on if the user is on an Apple silicon Mac or not while using `hatch` environments?

**Comment by ofek** [comment]
Do [environment markers](https://hatch.pypa.io/latest/config/dependency/#environment-markers) not work for this use case?

**Comment by matthewfeickert** [comment]
> Do [environment markers](https://hatch.pypa.io/latest/config/dependency/#environment-markers) not work for this use case?

Yes, yes they will. Thanks @ofek! (I actually _do_ generally read the docs for projects, but I also usually post these GitHub comments as reminders to do so when it isn't after midnight local time for me).

The way that `hatchling` does this is actually super nice, so many thanks for all the work that went into this design. :+1:

**Comment by matthewfeickert** [comment]
`hatchling` environment markers now enabling PR #2119. :rocket:
